src/server/debug/server/server.go

```diff
@@ -815,14 +831,25 @@ func (s *debugServer) forEachWorkerLoki(ctx context.Context, pipelineInfo *pps.P
 	return nil
 }
 
+// quoteLogQLStreamSelector returns a string quoted as a LogQL stream selector.
+// The rules for quoting a LogQL stream selector are documented at
+// https://grafana.com/docs/loki/latest/logql/log_queries/#log-stream-selector
+// to be the same as for a Prometheus string literal, as documented at
+// https://prometheus.io/docs/prometheus/latest/querying/basics/#string-literals.
+// This happens to be the same as a Go string, with single or double quotes or
+// backticks allowed as enclosing characters.
+func quoteLogQLStreamSelector(s string) string {
+	return fmt.Sprintf("%q", s)
```

Comment:
strconv.Quote

Apparently "github.com/prometheus/prometheus/model/labels" is the actual library used to parse these selectors; it prints them with strconv.Quote as well.  I suppose if we already depend on that library you could use that directly though.  (labels.NewMatcher(labels.Equal, "pipeline", whatever)).